### PR TITLE
TVB 2193

### PR DESCRIPTION
**Pull request by sipv**
Fixed indexing in SparseCoupling. Expanded unittest for Difference coupling. Small cleanup of CouplingTest setup.

More robust testing would be to compare the results of the same coupling based on SparseCoupling and Coupling classes and see if they are the equal.

### CODE REVIEW

File: tvb/simulator/coupling.py [hunk 1]
```diff
@@ -159,10 +159,18 @@ def _lri(self, nnz_row_el_idx):
     def __call__(self, step, history):
         h = history # type: SparseHistory
         x_i, x_j = h.query_sparse(step)
+        assert x_i.shape == (h.n_cvar, h.n_node, h.n_mode)
+        assert x_j.shape == (h.n_cvar, h.n_nnzw, h.n_mode)
+        #                              ^ from (columns)
+
         sum = numpy.zeros_like(x_i)
-        x_i = x_i[:, h.nnz_col_el_idx]
+        x_i = x_i[:, h.nnz_row_el_idx]
```
Comment (maedoc) [inline tvb/simulator/coupling.py:167]: Good catch, bad typo on my part.

**Review by maedoc — APPROVED** [review]

**Comment by maedoc** [comment]
👍 @sipv

**Comment by sipv** [comment]
Thanks. I guess the Jira issue can be closed now, though I am not sure if I can (or should) do it.

**Comment by maedoc** [comment]
yes, please mark as resolved (not closed) with a link to this PR.

**Pull request merged**